kernel/src/debug.rs

```diff
@@ -88,23 +88,51 @@ pub trait IoWrite {
 ///////////////////////////////////////////////////////////////////
 // panic! support routines
 
-/// Tock default panic routine.
+/// Tock panic routine, without the infinite LED-blinking loop.
+///
+/// This is useful for boards which do not feature LEDs to blink or
+/// want to implement their own behaviour. This method returns after
+/// performing the panic dump.
+///
+/// After this method returns, the system is no longer in a
+/// well-defined state. Care must be taken on how one interacts with
+/// the system once this function returns.
 ///
 /// **NOTE:** The supplied `writer` must be synchronous.
-pub unsafe fn panic<L: hil::led::Led, W: Write + IoWrite, C: Chip>(
-    leds: &mut [&L],
+pub unsafe fn panic_noloop<W: Write + IoWrite, C: Chip>(
```

Comment:
I think a better name might be `panic_print_all` or something.

Reply:
Applied the suggestion from Brad below, which is essentially the same. Hope that's okay.